en/_posts/2015-05-05-sailabove-documentation-predictor.markdown

```diff
@@ -0,0 +1,55 @@
+---
+layout: post
+title: "Working with the Predictor load balancer"
+categories: docker/documentation
+tags: sailabove, reference
+lang: en
+author: yadutaf
+---
+
+If you are building or running a web/HTTP based application, you can use the "predictor" to get out-of-the-box load-balancing. The predictor is based on technologies originally built for OVH's shared Hosting. It scales!
+
+When a service is configured to work with the predictor, the containers are automatically registered or unregistered as the service is scaled up or down. If a container stops responding, it will automatically be marked as faulty and further requests will be directed to another container.
+
+Technically, the predictor is built on top of the proven Nginx reverse proxy. It supports websockets and POST requests streaming so that there is no lag due to buggering neither upload limit. Unless you define one in your application, of course.
+
+# Getting started
+
+All services using ``--network predictor`` will automatically use the load balancer. For example, to start a hello world application on the predictor:
+
+```
+sail services add demo/hello-python <user name>/hello-python --network predictor
+```
+
+As customers value polite applications, we are now handling thousands of hello requests per second, so that we need to scale:
+
+```
+sail services scale <user name>/hello-python --number 3
+```
+
+That's all. All 3 containers are now handling their share of the requests on domain ``http://hello-python.<user name>.app.sailabove.io/``
+
+# Attaching custom domains
+
+Now that you have a scalable, load-balanced application, you will most probably want to publish it with an appealing domain. Let's say its ``www.sayhello.com``. All you need to do is:
+
+```
+sail services domain-attach <user name>/hello-python www.sayhello.com
+```
+
+It can be detached at any time with:
+
+```
+sail services domain-detach <user name>/hello-python www.sayhello.com
+```
+
```

Comment:
It might make sense to add some details about load balancing:

# Load-Balancing details

Once attached to predictor, incoming requests will be load-balanced in round-robin fashion for new clients. But what if I want to store some state on the container ?

After the first request, a *sticky cookie* will be used to always redirect that client to a particular container, if possible. So you can safely use your local `/tmp` for instance to store session-specific data.